tests/foreign_object/tests.py

```diff
@@ -412,3 +412,28 @@ def test_foreign_object_form(self):
         a2 = form.save()
         self.assertEqual(a.pk, a2.pk)
         self.assertEqual(a2.pub_date, datetime.date(2013, 1, 1))
+
+
+class RestrictedConditionsTests(TestCase):
+
+    def setUp(self):
+        for i in range(10):
+            Number.objects.create()
```

Comment:
This and below test assumes the primary key assignment starts from 1. This isn't necessarily true. It would be better to add an explicit "num" column to the Number model and assign 1 to it.